File: Data Structures & Algorithms/longest-consecutive-sequence/submission-6.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        nums = set(nums)
        res = 0
        for i in nums:
            curr = 1
            temp = i
            while temp-1 in nums:
                curr +=1
                temp = temp -1
            res = max(curr,res)
        return res

File: Data Structures & Algorithms/products-of-array-discluding-self/submission-6.cpp
class Solution {
public:
    vector<int> productExceptSelf(vector<int>& nums) {
        vector<int> res(nums.size(),1);
        int s = 1;
        for(int i = 0; i < nums.size()-1; i ++)
        {
            s = s * nums[i];
            res[i+1] = s;
        }
        s = 1;
        for(int i = nums.size()-1; i > 0; i --)
        {
            s = s * nums[i];
            res[i-1] = s * res[i-1];
        }
        return res;
    }
};
